api/server/router/network/filter.go

```diff
@@ -29,6 +29,29 @@ func filterNetworkByType(nws []types.NetworkResource, netType string) ([]types.N
 	return retNws, nil
 }
 
+func filterNetworkByUse(nws []types.NetworkResource, danglingOnly bool) ([]types.NetworkResource, error) {
+	retNws := []types.NetworkResource{}
+
+	switch danglingOnly {
+	case true:
+		for _, nw := range nws {
+			if len(nw.Containers) == 0 && !runconfig.IsPreDefinedNetwork(nw.Name) {
```

Comment:
My opinion is to add `DryRun` to `RemoveNetworkRequest`.
https://github.com/docker/swarmkit/blob/450df1ca02727e79b4c58e8448173150f3c56c25/api/control.proto#L289-L292

cc @aluzzardi @aaronlehmann